lib/elixir/lib/map_set.ex

```diff
@@ -121,7 +124,7 @@ defmodule MapSet do
       #MapSet<[1, 3]>
 
   """
-  @spec delete(t, value) :: t
+  @spec delete(t(val), val) :: t(val) when val: value
```

Comment:
Given what we have for intersection I think we should use `val1`, `val2` here as `val2` need not be in `val1`?